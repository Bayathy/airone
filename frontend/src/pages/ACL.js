import React from "react";
import { getACL } from "../utils/AironeAPIClient";
import AironeBreadcrumbs from "../components/common/AironeBreadcrumbs";
import Typography from "@material-ui/core/Typography";
import { Link, useParams } from "react-router-dom";
import { useAsync } from "react-use";
import ACLForm from "../components/common/ACLForm";

export default function ACL({}) {
  const { entityId } = useParams();

  const acl = useAsync(async () => {
    return getACL(entityId);
  });

  return (
    <div className="container-fluid">
      <AironeBreadcrumbs>
        <Typography component={Link} to="/new-ui/">
          Top
        </Typography>
        <Typography color="textPrimary">ACL</Typography>
      </AironeBreadcrumbs>

      {!acl.loading && (
        <>
          <Typography>{acl.value.object.name} の ACL 設定</Typography>
          <ACLForm acl={acl.value} />
        </>
      )}
    </div>
  );
}
