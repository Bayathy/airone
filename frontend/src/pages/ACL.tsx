import { Box, Typography } from "@mui/material";
import React, { FC } from "react";
import { Link, useParams } from "react-router-dom";
import { useAsync } from "react-use";

import { topPath } from "Routes";
import { aironeApiClientV2 } from "apiclient/AironeApiClientV2";
import { ACLForm } from "components/common/ACLForm";
import { AironeBreadcrumbs } from "components/common/AironeBreadcrumbs";
import { Loading } from "components/common/Loading";

export const ACL: FC = () => {
  const { entityId } = useParams<{ entityId: number }>();

  const acl = useAsync(async () => {
    return await aironeApiClientV2.getAcl(entityId);
  });

  return (
    <Box className="container-fluid">
      <AironeBreadcrumbs>
        <Typography component={Link} to={topPath()}>
          Top
        </Typography>
        <Typography color="textPrimary">ACL</Typography>
      </AironeBreadcrumbs>

      {acl.loading ? (
        <Loading />
      ) : (
        <>
          <Typography>{acl.value.name} の ACL 設定</Typography>
          <ACLForm objectId={entityId} acl={acl.value} />
        </>
      )}
    </Box>
  );
};
