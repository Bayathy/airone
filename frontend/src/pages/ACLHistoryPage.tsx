import { Box, Container } from "@mui/material";
import React, { FC, useEffect, useState } from "react";
import { useAsync } from "react-use";

import { ACLObjtypeEnum } from "apiclient/autogenerated";
import { ACLHistoryList } from "components/acl/ACLHistoryList";
import { Loading } from "components/common/Loading";
import { PageHeader } from "components/common/PageHeader";
import { EntityBreadcrumbs } from "components/entity/EntityBreadcrumbs";
import { EntryBreadcrumbs } from "components/entry/EntryBreadcrumbs";
import { useTypedParams } from "hooks/useTypedParams";
import { aironeApiClientV2 } from "repository/AironeApiClientV2";

export const ACLHistoryPage: FC = () => {
  const { objectId } = useTypedParams<{ objectId: number }>();
  const [breadcrumbs, setBreadcrumbs] = useState<JSX.Element>(<Box />);

  const acl = useAsync(async () => {
    return await aironeApiClientV2.getAcl(objectId);
  }, [objectId]);

  const aclHistory = useAsync(async () => {
    return await aironeApiClientV2.getAclHistory(objectId);
  }, [objectId]);

  useEffect(() => {
    if (acl.value == null) return;

    switch (acl.value.objtype) {
      case ACLObjtypeEnum.Entity:
        aironeApiClientV2.getEntity(objectId).then((resp) => {
          setBreadcrumbs(
            <EntityBreadcrumbs entity={resp} title="ACL変更履歴" />
          );
        });
        break;
      case ACLObjtypeEnum.Entry:
        aironeApiClientV2.getEntry(objectId).then((resp) => {
          setBreadcrumbs(<EntryBreadcrumbs entry={resp} title="ACL変更履歴" />);
        });
        break;
    }
  }, [acl]);

  return (
    <Box className="container-fluid">
      {breadcrumbs}

      <PageHeader title={acl.value?.name ?? ""} description="ACL変更履歴" />

      {aclHistory.loading ? (
        <Loading />
      ) : (
        <Container>
          <ACLHistoryList histories={aclHistory.value ?? []} />
        </Container>
      )}
    </Box>
  );
};
