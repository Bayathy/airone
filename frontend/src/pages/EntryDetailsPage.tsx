import AppsIcon from "@mui/icons-material/Apps";
import ArrowDropDownIcon from "@mui/icons-material/ArrowDropDown";
import {
  Box,
  Chip,
  Container,
  Grid,
  IconButton,
  Stack,
  Typography,
} from "@mui/material";
import React, { FC, useState } from "react";
import { Link, useParams } from "react-router-dom";
import { useAsync, useLocation } from "react-use";

import { entitiesPath, entityEntriesPath, topPath } from "Routes";
import { aironeApiClientV2 } from "apiclient/AironeApiClientV2";
import { AironeBreadcrumbs } from "components/common/AironeBreadcrumbs";
import { Loading } from "components/common/Loading";
import { EntryAttributes } from "components/entry/EntryAttributes";
import { EntryControlMenu } from "components/entry/EntryControlMenu";
import { EntryReferral } from "components/entry/EntryReferral";

export const EntryDetailsPage: FC = () => {
  const { entityId } = useParams<{ entityId: number }>();
  const { entryId } = useParams<{ entryId: number }>();

  const [entryAnchorEl, setEntryAnchorEl] =
    useState<HTMLButtonElement | null>();

  const location = useLocation();

  const refAttrList = React.createRef<HTMLSpanElement>();

  React.useEffect(() => {
    switch (location.hash) {
      case "#attr_list":
        refAttrList.current.scrollIntoView({
          behavior: "smooth",
          block: "start",
        });
    }
  }, [location.hash]);

  const entry: any = useAsync(async () => {
    return await aironeApiClientV2.getEntry(entryId);
  }, [entryId]);

  return (
    <Box display="flex" flexDirection="column" flexGrow="1">
      <AironeBreadcrumbs>
        <Typography component={Link} to={topPath()}>
          Top
        </Typography>
        <Typography component={Link} to={entitiesPath()}>
          エンティティ一覧
        </Typography>
        {!entry.loading && (
          <Typography
            component={Link}
            to={entityEntriesPath(entry.value.schema.id)}
          >
            {entry.value.schema.name}
          </Typography>
        )}
        {!entry.loading && (
          <Typography color="textPrimary">{entry.value.name}</Typography>
        )}
      </AironeBreadcrumbs>

      <Container maxWidth="lg" sx={{ pt: "112px" }}>
        <Box display="flex">
          <Box width="50px" />
          <Box flexGrow="1">
            {!entry.loading && (
              <Typography variant="h2" align="center">
                {entry.value.name}
              </Typography>
            )}
            <Typography variant="h4" align="center">
              エントリ詳細
            </Typography>
          </Box>
          <Box width="50px">
            <IconButton
              onClick={(e) => {
                setEntryAnchorEl(e.currentTarget);
              }}
            >
              <AppsIcon />
            </IconButton>
            <EntryControlMenu
              entryId={entryId}
              anchorElem={entryAnchorEl}
              handleClose={() => setEntryAnchorEl(null)}
            />
          </Box>
        </Box>
      </Container>
      <Stack
        direction="row"
        spacing={1}
        sx={{ justifyContent: "center", pt: "16px", pb: "64px" }}
      >
        <Chip
          icon={<ArrowDropDownIcon />}
          label="項目一覧"
          clickable={true}
          variant="outlined"
          component={Link}
          to="#attr_list"
          sx={{
            flexDirection: "row-reverse",
            "& span": {
              pr: "0px",
            },
            "& svg": {
              pr: "8px",
            },
          }}
        />
      </Stack>

      <Grid
        container
        flexGrow="1"
        columns={6}
        sx={{ borderTop: 1, borderColor: "#0000008A" }}
      >
        <Grid
          item
          xs={1}
          sx={{
            py: "64px",
            borderRight: 1,
            borderColor: "#0000008A",
          }}
        >
          <EntryReferral entityId={entityId} entryId={entryId} />
        </Grid>
        <Grid item xs={4}>
          <Box p="32px">
            <Typography
              p="32px"
              fontSize="32px"
              align="center"
              ref={refAttrList}
            >
              項目一覧
            </Typography>
            {entry.loading ? (
              <Loading />
            ) : (
              <EntryAttributes attributes={entry.value.attrs} />
            )}
          </Box>
        </Grid>
        <Grid item xs={1} />
      </Grid>
    </Box>
  );
};
