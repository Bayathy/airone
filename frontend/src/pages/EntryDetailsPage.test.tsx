/**
 * @jest-environment jsdom
 */

import {
  render,
  waitForElementToBeRemoved,
  screen,
} from "@testing-library/react";
import React from "react";

import { EntryDetailsPage } from "pages/EntryDetailsPage";
import { TestWrapper } from "utils/TestWrapper";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  const entry = {
    id: 1,
    name: "aaa",
    schema: {
      id: 2,
      name: "bbb",
    },
    attrs: [],
  };

  const referredEntries = {
    entries: [
      {
        id: 1,
        name: "aaa",
        entity: "bbb",
      },
    ],
  };

  // wait async calls and get rendered fragment
  const result = render(<EntryDetailsPage />, {
    wrapper: TestWrapper,
  });
  await waitForElementToBeRemoved(screen.getByTestId("loading"));

  expect(result).toMatchSnapshot();
});
