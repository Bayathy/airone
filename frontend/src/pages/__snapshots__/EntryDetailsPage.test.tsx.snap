// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1ylu0bo"
      >
        <div
          class="MuiBox-root css-1l11j2j"
        >
          <div
            class="makeStyles-wrapper-1 MuiBox-root css-0"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
            >
              <nav
                aria-label="breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/new-ui/"
                    >
                      Top
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/new-ui/entities"
                    >
                      エンティティ一覧
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                      href="/new-ui/entities/2/entries"
                    >
                      bbb
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    /
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                    >
                      aaa
                    </p>
                  </li>
                </ol>
              </nav>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-ca23fy"
        />
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-m0xtue-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-k008qs"
          >
            <div
              class="MuiBox-root css-kgitkj"
            />
            <div
              class="MuiBox-root css-i9gxme"
            >
              <h2
                class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
              >
                aaa
              </h2>
              <h4
                class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
              >
                エントリ詳細
              </h4>
            </div>
            <div
              class="MuiBox-root css-kgitkj"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AppsIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="css-1p01se3-MuiStack-root"
        >
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1n5jo98-MuiButtonBase-root-MuiChip-root"
            role="button"
            tabindex="0"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m7 10 5 5 5-5z"
              />
            </svg>
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            >
              項目一覧
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-3g06bi-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-fpette-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-gajhq5"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-xtv7g-MuiTypography-root"
                >
                  関連づけられたエントリ(計
                  1
                  )
                </p>
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-3wf0vq-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="SearchIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                        />
                      </svg>
                    </div>
                    <input
                      aria-invalid="false"
                      class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-1"
                      placeholder="エントリを絞り込む"
                      type="text"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-nnbavb"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
              <ul
                class="MuiList-root MuiList-padding css-1dto0w0-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters MuiListItem-divider css-1wa7a4p-MuiListItem-root"
                >
                  <a
                    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                    href="/new-ui/entities/undefined/entries/1/details"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemText-root css-r83ncu-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                      >
                        aaa
                      </span>
                    </div>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </a>
                </li>
              </ul>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-12dpnmv-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-1wcaknn"
            >
              <div
                name="attr_list"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-lr5qxb-MuiTypography-root"
              >
                項目一覧
              </p>
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
              >
                <table
                  class="MuiTable-root css-rqglhn-MuiTable-root"
                >
                  <thead
                    class="MuiTableHead-root css-15n4dzy-MuiTableHead-root"
                  >
                    <tr
                      class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                    >
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-iptfz6-MuiTableCell-root"
                        scope="col"
                      >
                        項目
                      </th>
                      <th
                        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-iptfz6-MuiTableCell-root"
                        scope="col"
                      >
                        内容
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                  />
                </table>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-a0vo9r-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1ylu0bo"
    >
      <div
        class="MuiBox-root css-1l11j2j"
      >
        <div
          class="makeStyles-wrapper-1 MuiBox-root css-0"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
          >
            <nav
              aria-label="breadcrumb"
              class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-63vwl1-MuiTypography-root-MuiBreadcrumbs-root"
            >
              <ol
                class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
              >
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    href="/new-ui/"
                  >
                    Top
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    href="/new-ui/entities"
                  >
                    エンティティ一覧
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <a
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                    href="/new-ui/entities/2/entries"
                  >
                    bbb
                  </a>
                </li>
                <li
                  aria-hidden="true"
                  class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                >
                  /
                </li>
                <li
                  class="MuiBreadcrumbs-li"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-1gqyz35-MuiTypography-root"
                  >
                    aaa
                  </p>
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root css-ca23fy"
      />
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-m0xtue-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiBox-root css-kgitkj"
          />
          <div
            class="MuiBox-root css-i9gxme"
          >
            <h2
              class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-icizyq-MuiTypography-root"
            >
              aaa
            </h2>
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-149xwtl-MuiTypography-root"
            >
              エントリ詳細
            </h4>
          </div>
          <div
            class="MuiBox-root css-kgitkj"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AppsIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="css-1p01se3-MuiStack-root"
      >
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-clickable MuiChip-clickableColorDefault MuiChip-outlinedDefault css-1n5jo98-MuiButtonBase-root-MuiChip-root"
          role="button"
          tabindex="0"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconMedium MuiChip-iconColorDefault css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m7 10 5 5 5-5z"
            />
          </svg>
          <span
            class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
          >
            項目一覧
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-3g06bi-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-fpette-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-gajhq5"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-xtv7g-MuiTypography-root"
              >
                関連づけられたエントリ(計
                1
                )
              </p>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-3wf0vq-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-1"
                    placeholder="エントリを絞り込む"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-nnbavb"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <ul
              class="MuiList-root MuiList-padding css-1dto0w0-MuiList-root"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-divider css-1wa7a4p-MuiListItem-root"
              >
                <a
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                  href="/new-ui/entities/undefined/entries/1/details"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-r83ncu-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      aaa
                    </span>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-12dpnmv-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1wcaknn"
          >
            <div
              name="attr_list"
            />
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-lr5qxb-MuiTypography-root"
            >
              項目一覧
            </p>
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
            >
              <table
                class="MuiTable-root css-rqglhn-MuiTable-root"
              >
                <thead
                  class="MuiTableHead-root css-15n4dzy-MuiTableHead-root"
                >
                  <tr
                    class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  >
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-iptfz6-MuiTableCell-root"
                      scope="col"
                    >
                      項目
                    </th>
                    <th
                      class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-iptfz6-MuiTableCell-root"
                      scope="col"
                    >
                      内容
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                />
              </table>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-a0vo9r-MuiGrid-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
